### DISCUSSION TITLE
Mapsforge map features, mapsforge as tile provider

### DISCUSSION THREAD

**Discussion by 8Duke8** [mikes222/mapsforge_flutter#64] [General]
Hi @mikes222,

First of all, thanks for the great job you're doing for the community. I am evaluating mapsForge maps as an offline map provider for my app. 

I want to ask several questions. 

1) Does the map support showing jpg images (2000x2000) as markers? I need to show detailed maps (downloaded jpgs ~700kB each) over the mapsforge offline map. As I tried and also looked into the code only .svg and .png files are supported.

2) Can I somehow move the marker on the map (ideally by tapping and dragging the marker over the map)? It's a must-have feature for me. I didn't find this functionality in the documentation and source codes.

3) If the map doesn't support the above features, is there an easy way, how I could create a tile provider from the mapsforge offline map sources? In fact, my application already uses online flutter-maps (https://pub.dev/packages/flutter_map), but I need a full offline vector map (mapsforge sources https://download.mapsforge.org/maps/v5/europe/ are great for this purpose). If I could create an offline tile provider that would return vector maps in a format like here https://github.com/greensopinion/flutter-vector-map-tiles. If you could point me in any direction, I would most appreciate it.

**Comment by mikes222** [comment]
ad 1: Markers are meant for POIs like restaurants or fuel stations or for distinct (relative) small areas like rectangle, circle or polygons. I suggest to use an overlay instead which is nothing more than a plain old flutter widget (pofw :-) ) which gets rebuilt whenever the position or zoom changes. 

In case you want to support jpg for markers anyway just look at SymbolCache and its used classes. Here you can add support for jpg too. 

ad 2: Drag'n'drop is currently not implemented. It needs to be done in the ``FlutterGestureDetector`` and its event-class. We need this funtionality anyway so feel free to play around with it and create a Pull Request.

ad 3: We have already several tile providers like ``ArcGisOnlineRenderer``. If I understood your request correctly you want to store the european XML from OSM directly at the user's device and render the maps from there. Think of it again since the XML is huge and the render process will be painfully slow. Thats why mapfiles have been developed.
